src/compiler/checker.ts

```diff
@@ -5163,8 +5167,8 @@ namespace ts {
             let result: TypeParameter[];
             for (const node of symbol.declarations) {
                 if (node.kind === SyntaxKind.InterfaceDeclaration || node.kind === SyntaxKind.ClassDeclaration ||
-                    node.kind === SyntaxKind.ClassExpression || node.kind === SyntaxKind.TypeAliasDeclaration) {
-                    const declaration = <InterfaceDeclaration | TypeAliasDeclaration>node;
+                    node.kind === SyntaxKind.ClassExpression || node.kind === SyntaxKind.TypeAliasDeclaration || node.kind === SyntaxKind.JSDocTypedefTag) {
+                    const declaration = <InterfaceDeclaration | TypeAliasDeclaration | JSDocTypedefTag>node;

```

Comment:
same. There is a predicate `isJSDocTypeAlias` that should maybe just be `isTypeAlias` and include SyntaxKind.TypeAliasDeclaration.

Reply:
https://github.com/Microsoft/TypeScript/pull/23947 isn't merged yet

Reply:
https://github.com/Microsoft/TypeScript/pull/23947 isn't merged yet